### PR TITLE
nimble/host: move att_truncate_to_mtu to ble_att_svr_tx_rsp

### PR DESCRIPTION
**Pull request by piotrnarajowski**

### CODE REVIEW

**Comment by piotrnarajowski** [comment]
#AutoPTS run mynewt GATT/SR --test_case_limit 10

**Comment by piotrnarajowski** [comment]
@rymanluk please review

File: nimble/host/src/ble_att_svr.c [hunk 1]
```diff
@@ -636,29 +638,43 @@ ble_att_svr_tx_rsp(uint16_t conn_handle, uint16_t cid, int hs_status, struct os_
...
+            }
+
+            ble_att_truncate_to_mtu(chan, om);
             hs_status = ble_att_tx(conn_handle, cid, om);
             om = NULL;
             if (hs_status) {
                 err_status = BLE_ATT_ERR_UNLIKELY;
             }
         }
+    }
```
Comment (KKopyscinski) [inline nimble/host/src/ble_att_svr.c:660]: Continuation no longer part of ` if (do_tx) {` - I don't think it's intentional?

File: nimble/host/src/ble_att_svr.c [hunk 2]
```diff
@@ -636,29 +638,43 @@ ble_att_svr_tx_rsp(uint16_t conn_handle, uint16_t cid, int hs_status, struct os_
...
             om = NULL;
             if (hs_status) {
                 err_status = BLE_ATT_ERR_UNLIKELY;
             }
         }
 
-        if (hs_status != 0) {
-            STATS_INC(ble_att_stats, error_rsp_tx);
+done:
+    if (hs_status != 0) {
```
Comment (KKopyscinski) [inline nimble/host/src/ble_att_svr.c:662]: nitpick: indentation (one level deeper, as it was), from here to end

Comment (piotrnarajowski) [reply nimble/host/src/ble_att_svr.c]: Should be ok now

File: nimble/host/src/ble_att_svr.c (hunk 1 above)
Comment (piotrnarajowski) [reply nimble/host/src/ble_att_svr.c]: Nope, missed it

**Review by rymanluk — APPROVED** [review]

**Review by rymanluk — COMMENTED** [review]
Thanks, looks good to me

**Review by KKopyscinski — APPROVED** [review]

**Review by sjanc — CHANGES_REQUESTED** [review]

File: nimble/host/src/ble_att_svr.c [hunk 3]
```diff
@@ -636,13 +638,30 @@ ble_att_svr_tx_rsp(uint16_t conn_handle, uint16_t cid, int hs_status, struct os_
 
     if (do_tx) {
         if (hs_status == 0) {
+            ble_hs_lock();
+            conn = ble_hs_conn_find(conn_handle);
+            if (conn == NULL) {
+                hs_status = BLE_HS_ENOTCONN;
+                goto done;
```
[the remark this answers is not in the document]
Comment (piotrnarajowski) [reply nimble/host/src/ble_att_svr.c:646]: Fixed

[the remark this answers is not in the document]
Comment (piotrnarajowski) [reply nimble/host/src/ble_att_svr.c:646]: Fixed

**Review by sjanc — APPROVED** [review]

**Pull request merged**